### ISSUE TITLE
bug in neighborhood function

### ISSUE THREAD

**Issue by alimamdouh212** [JuliaGraphs/LightGraphs.jl#1363]
```
g = SimpleGraph(3)
add_edge!(g, 1, 2)
add_edge!(g, 1, 3)
add_edge!(g, 3, 2)
dismx = zeros((3, 3))
dismx[1, 2] = 10
dismx[1, 3] = 1
dismx[3, 2] = 1
neighborhood(g, 1, 4, dismx)

```
will result in `[1, 3]` despite 2 can be reached from 1 by a path of length 2 .

**Comment by sbromberger** [comment]
Possibly related to #1116

**Comment by alimamdouh212** [comment]
no, the issue in  #1116 no longer exists.

**Comment by sbromberger** [comment]
I know the issue no longer exists, since it was closed in #1166, but the fix that we used there may be the cause or otherwise related.

**Comment by alimamdouh212** [comment]
The cause of this issue, is that with the first vist to any node, it
considered visited, and its distance cannot be modified, even if there is
some other path that have shorter distance, but haven't been discovered
yet, the only way to address this isuue is by using Dijkstra's algorithm.

**Comment by alimamdouh212** [comment]
Can I submit a PR for it?

**Comment by sbromberger** [comment]
It will be fixed once we merge #1332 and migrate neighborhood to use the new shortest paths / traversal code.
